reading_scraper.py

```diff
@@ -29,115 +53,173 @@ def useragent_generator():
     # print(headers)
     return headers
 
-full_headers = {
+def check_if_pdf(url):
+    is_pdf = re.findall(r".pdf", url, flags=re.IGNORECASE)
+    # print(is_pdf)
+    if len(is_pdf) > 0:
+        # print("it's a pdf")
+        # go to pdf scraper
+        return True
+    else:
+        return False
+
+def get_reading_data(url, cached_url):
+    global reading, title, description, image
+
+    full_headers = {
     'Accept': '*/*',
     'Accept-Encoding': 'gzip, deflate, br',
     'Accept-Language': 'en-US,en;q=0.9',
     'content-Type': 'application/x-www-form-urlencoded; charset=UTF-8',
     'Cache-Control': 'no-cache',
     'DNT': '1',
     'Pragma': 'no-cache',
-    'Referer': BASE_URL,
+    'Referer': "https://www.google.com/search",
     'User-Agent': useragent_generator(),
     'Sec-Fetch-Dest': 'document',
     'Sec-Fetch-Mode': 'navigate',
     'Sec-Fetch-Site': 'cross-site',
     'Upgrade-Insecure-Requests': '1',
-}
-
-def get_reading_data(url, cached_url):
-    global reading, title, description, image, special_sites, full_headers
+    }
 
+    
+    g = Goose({'http_headers': full_headers})
+    # g = Goose({'browser_user_agent': useragent_generator()})
+    # extract info with goose
     try:
-        g = Goose({'http_headers': full_headers})
-        # g = Goose({'browser_user_agent': useragent_generator()})
-        # extract info with goose
         reading = g.extract(url = url)
+    except:
+        print("Error occurred with 1st try getting reading data: ", sys.exc_info())
+        reading = ''
 
-        # if domain is 'special' or if title is blank, try cached version
-        if (reading.title == '' 
-            or '403 Forbidden' in reading.title 
-            or any(domain in url for domain in special_sites)):
-            # print('needing to cache')
-            # update header to refer from google
-            full_headers = {
-                'Accept': '*/*',
-                'Accept-Encoding': 'gzip, deflate, br',
-                'Accept-Language': 'en-US,en;q=0.9',
-                'content-Type': 'application/x-www-form-urlencoded; charset=UTF-8',
-                'Cache-Control': 'no-cache',
-                'DNT': '1',
-                'Pragma': 'no-cache',
-                'Referer': 'https://www.google.com/',
-                'User-Agent': useragent_generator(),
-                'Sec-Fetch-Dest': 'document',
-                'Sec-Fetch-Mode': 'navigate',
-                'Sec-Fetch-Site': 'cross-site',
-                'Upgrade-Insecure-Requests': '1',
-            }
+    # if domain is 'special' or if title is blank, try cached version
+    if (reading == '' or reading.title == '' 
+        or '403 Forbidden' in reading.title
+        or "Bloomberg" in reading.title): 
+        # or any(domain in url for domain in special_sites)):
+        print('needing to cache')
+        # update header to refer from google
+        full_headers = {
```

Comment:
Let's place `full_headers` in its own function so it is consistent across the file